### ISSUE TITLE
[Bug] Multi-class probability outputs incorrect

### ISSUE THREAD

**Issue by RAMitchell** [triton-inference-server/fil_backend#343]
An xgboost model with 3 classes and predict_proba=true returns incorrect output.

To fix:

The softmax output type needs to be selected in the multiclass case, like max index is selected here:
https://github.com/triton-inference-server/fil_backend/blob/956a03b9def480b7710cd2fea1f25afdf2885120/src/tl_model.h#L45

The softmax implementation also has a mistake in its indexing:
https://github.com/triton-inference-server/fil_backend/blob/956a03b9def480b7710cd2fea1f25afdf2885120/src/herring/model.hpp#L186
Should read `auto const row_begin = output + row_index * num_class;`

Also here:
https://github.com/triton-inference-server/fil_backend/blob/956a03b9def480b7710cd2fea1f25afdf2885120/src/herring/model.hpp#L199
